File: 2176-count-equal-and-divisible-pairs-in-an-array/2176-count-equal-and-divisible-pairs-in-an-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var countPairs = function(nums, k) {
  const l = nums.length
  let ans = 0
  for (let i = 0; i < l - 1; i ++) {
    for (let j = i + 1; j < l; j++) {
      if ((i*j % k) === 0 && nums[i] === nums[j]) {
        ans ++
      }
    }
  }
  return ans
};

File: 2176-count-equal-and-divisible-pairs-in-an-array/2176-count-equal-and-divisible-pairs-in-an-array.go
func countPairs(nums []int, k int) int {
  ans := 0
  for i := 0; i < len(nums) - 1; i++ {
    vi := nums[i]
    for j:= i+1; j < len(nums); j++ {
      if vi == nums[j] && (i*j)%k == 0 {
        ans++
      }
    }
  }
  return ans
}